types/mjml-core/index.d.ts

```diff
@@ -0,0 +1,256 @@
+     */
+    fonts?: { [key: string]: string };
+
+    /**
+     * Option to keep comments in the HTML output
+     * default: true
+     */
+    keepComments?: boolean;
+
+    /**
+     * @deprecated use js-beautify directly after processing the MJML
+     *
+     * Option to beautify the HTML output
+     * default: false
+     */
+    beautify?: boolean;
+
+    /**
+     * @deprecated use html-minifier directly after processing the MJML
+     *
+     * Option to minify the HTML output
+     *
+     * default: false
+     */
+    minify?: boolean;
+    /**
+     * @deprecated @see minify
+     *
+     * Options for html minifier, see mjml-cli documentation for more info
+     * Passed directly to html-minifier as options
+     *
+     * default: @see htmlMinify usage in mjml-core/src/index.js
+     */
+    minifyOptions?: MJMLMinifyOptions;
+
+    /**
+     * How to validate your MJML
+     *
+     * skip: your document is rendered without going through validation
+     * soft: your document is going through validation and is rendered, even if it has errors
+     * strict: your document is going through validation and is not rendered if it has any error
+     *
+     * default: soft
+     */
+    validationLevel?: 'strict' | 'soft' | 'skip';
+
+    /**
+     * Full path of the specified file to use when resolving paths from mj-include components
+     * default: '.'
+     */
+    filePath?: string;
+    /**
+     * The path or directory of the .mjmlconfig file
+     * default: process.cwd()
+     */
+    mjmlConfigPath?: string;
+
+    /**
+     * Use the config attribute defined in the .mjmlconfig file.
+     * The config passed into mjml2html overrides the .mjmlconfig.
+     * default: false
+     */
+    useMjmlConfigOptions?: boolean;
+
+    /**
+     * optional setting when inlining css, see mjml-cli documentation for more info
+     */
+    juicePreserveTags?: { [index: string]: { start: string; end: string } };
+    juiceOptions?: any;
+
+    /**
+     * undocumented
+     * a function returning html skeleton
+     * default: see mjml-core/src/helpers/skeleton.js
+     */
+    skeleton?: string | (() => string);
+
+    actualPath?: string;
+    /**
+     * undocumented
+     * ignore mj-include elements
+     * default: false
+     */
+    ignoreIncludes?: any;
+    /**
+     * see mjml-parser-xml
+     */
+    preprocessors?: Array<((xml: string) => string)>;
+}
+
+export interface MJMLMinifyOptions {
+    collapseWhitespace?: boolean;
+    minifyCSS?: boolean;
+    removeEmptyAttributes?: boolean;
+}
+
+export interface MJMLParseResults {
+    html: string;
+    json: MJMLJsonObject;
+    errors: MJMLParseError[];
+}
+
+export interface MJMLParseError {
+    line: number;
+    message: string;
+    tagName: string;
+    formattedMessage: string;
+}
+
+export interface MJMLJsonObject {
+    tagName: string;
+    attributes: object;
+    children?: MJMLJsonObject[];
```

Comment:
I think we can type this a bit better
Either: 
- no children/no content (ex mj image)
- children but no content
- content but no children

Reply:
great suggestion! I'll do that